File: src/main/java/com/example/chatService/entity/PrivateMessage.java
package com.example.chatService.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;

@Entity
@Table(name = "private_message")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class PrivateMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String sender;    // username (또는 이메일, 프로젝트 규약에 맞게)

    @Column(nullable = false)
    private String receiver;  // username

    @Column(nullable = false, length = 2000)
    private String content;

    @Column(nullable = false)
    private OffsetDateTime sentAt;

    private boolean readFlag = false;
}

File: src/main/java/com/example/chatService/entity/PrivateMessage.kt
package com.example.chatService.entity

import jakarta.persistence.*
import java.time.OffsetDateTime

@Entity
@Table(name = "private_message")
class PrivateMessage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @Column(nullable = false)
    var sender: String = "",
    @Column(nullable = false)
    var receiver: String = "",
    @Column(nullable = false, length = 2000)
    var content: String = "",
    @Column(nullable = false)
    var sentAt: OffsetDateTime = OffsetDateTime.now(),
    var readFlag: Boolean = false
)
